jobserv_worker.py

```diff
@@ -713,6 +720,11 @@ def get_args(args=None):
         help="""Worker name to register. If none is provided, the value of
                 /etc/hostname will be used.""",
     )
+    p.add_argument(
+        "--insecure-ssl",
+        action="store_true",
+        help="Don't verify SSL connections to server. Handy in dev modes where TLS may be self-singed. Default=%(default)s",
+    )
```

Comment:
I guess this is really for the dev-mode and not to use in production/staging, right?

In general, I'd prefer if we have a configuration option to add a trusted CA.
In this case, a user generates a self-signed CA and configures jobserv to trust it.
But then, the API certificate should be still signed by that CA i.e. we don't need the insecure/no-verify flags.

What do you think?

Reply:
I'm using it in staging right now. I've been fighting python3 requrests/certifi/ssl to actually trust a LetsEncrypt staging cert. I get a system where Curl and Wget trust it but requests (and pure python ssl connect) won't.

I'll give it one more chance and see if I can make this work. Like you said - its more secure. It also means you don't have to change *every* place that makes an HTTP request.

Reply:
IIRC we have 3 options:
- use global trusted CA storage. By IIRC Python has some problems with that as it uses its own bundle.
- update every request just like you did with some `ca-xyz-file argument`.
- amend the global `requests` session object.

Reply:
See here for an example:
https://github.com/foundriesio/test-dspl/blob/main/users/device.py#L101
```
        self.client.cert = self.certfile, self.keyfile
        self.client.verify = self.cafile
```
IIUC you only need to set the `verify` attribute.

There `client` is the `requests.Session` object; but I _believe_ you can pass the same attribute to the `requests.request` function.

Reply:
> use global trusted CA storage. By IIRC Python has some problems with that as it uses its own bundle.

You can override this now and tell it to use something like `/etc/ssl/certs/ca-bundle.crt` with the environment variable `REQUESTS_CA_BUNDLE`. That just doesn't actually work when I try it. I'll keep debugging that - its "the right way".